velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -436,6 +437,70 @@ TEST_F(ArithmeticTest, isNanDouble) {
   EXPECT_EQ(false, isNan(std::nullopt));
 }
 
+TEST_F(ArithmeticTest, normalizeNanFloat) {
+  const auto normalizeNan = [&](std::optional<float> a) {
+    return evaluateOnce<float>("normalize_nan(c0)", a);
+  };
+
+  const auto hasher = [&](std::optional<float> f) {
+    return folly::hasher<float>()(f.value());
+  };
+
+  const auto toInt64 = [&](std::optional<float> f) -> int64_t {
+    return f.value();
+  };
+
+  EXPECT_EQ(toInt64(0.0f), toInt64(normalizeNan(0.0f)));
+  EXPECT_EQ(toInt64(0.0f), toInt64(normalizeNan(-0.0f)));
+  EXPECT_EQ(toInt64(1.0f), toInt64(normalizeNan(1.0f)));
+  // folly::hasher<float>()(kNan) != folly::hasher<float>()(0.0f / 0.0f)
+  // toInt64(kNan) != toInt64(0.0f / 0.0f)
+  EXPECT_EQ(toInt64(normalizeNan(kNan)), toInt64(normalizeNan(kNan)));
+  EXPECT_EQ(toInt64(normalizeNan(kNan)), toInt64(normalizeNan(0.0f / 0.0f)));
```

Comment:
I believe 0.0f / 0.0f and kInf - kInf produce kNan, so they are already the same and we are not really testing the function. Try changing function implementation to always return 'x' unmodified and see if the test fails. It might be better to use `std::numeric_limits<float>::signaling_NaN()` or make some nan by setting some of the first 23 bits.

Reply:
Updated, thanks.